### PR TITLE
[Fix](Nereids)fix nereids is failed to parse union with query covered parenthesis

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

sql like the format (q1, q2, q3 is a query): 
``` sql
(q1) 
UNION ALL (q2)
UNION ALL (q3)
ORDER BY keys
```
cannot be parsed by nereids, because order will be recognized as an alias of query, we add queryOrganization to avoid it.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -56,15 +56,17 @@ planType
...
     : {!doris_legacy_SQL_syntax}? cte? queryTerm queryOrganization
-    | {doris_legacy_SQL_syntax}? queryTerm
+    | {doris_legacy_SQL_syntax}? queryTerm queryOrganization
     ;
 
 queryTerm
     : queryPrimary                                                                       #queryTermDefault
     | left=queryTerm operator=(UNION | EXCEPT | INTERSECT)
-      setQuantifier? right=queryTerm                                                     #setOperation
+      setQuantifier? right=queryTerm                                                    #setOperation
```
Comment (englefly) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:69]: it seems that in the edit version the comments are not aligned

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSetOperation.java [hunk 2]
```diff
@@ -114,12 +114,13 @@ public List<NamedExpression> buildNewOutputs(List<Expression> leftCastExpression
...
+                Cast cast = ((Cast) expression);
                 newOutputs.add(new SlotReference(
-                        ((Cast) expression).child().toSql(), expression.getDataType(),
-                        ((Cast) expression).child().nullable()));
+                        cast.child().toSql(), expression.getDataType(),
+                        cast.child().nullable()));
             } else if (expression instanceof Slot) {
-                newOutputs.add(new SlotReference(
-                        expression.toSql(), expression.getDataType(), expression.nullable()));
+                Slot slot = ((Slot) expression);
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSetOperation.java:122]: is there any diff?

File: regression-test/suites/nereids_p0/join/test_join.groovy [hunk 3]
```diff
@@ -398,12 +398,12 @@ suite("test_join", "nereids_p0") {
...
-    sql"""select a.k1 k1, a.k2, a.k3, b.k1, b.k2, b.k3 from ${tbName1} a full outer join ${tbName2} b 
-             on a.k1 = b.k1 and a.k2 > b.k2 order by isnull(k1), 1, 2, 3, 4, 5 limit 65535"""
-    sql"""select a.k1 k1, a.k2, a.k3, b.k1, b.k2, b.k3 from ${tbName1} a left outer join ${tbName2} b 
+    sql"""select a.k1 k, a.k2, a.k3, b.k1, b.k2, b.k3 from ${tbName1} a full outer join ${tbName2} b 
+             on a.k1 = b.k1 and a.k2 > b.k2 order by isnull(k), 1, 2, 3, 4, 5 limit 65535"""
+    sql"""select a.k1 k, a.k2, a.k3, b.k1, b.k2, b.k3 from ${tbName1} a left outer join ${tbName2} b 
              on a.k1 = b.k1 and a.k2 > b.k2 union (select a.k1, a.k2, a.k3, b.k1, b.k2, b.k3 
              from ${tbName1} a right outer join ${tbName2} b on a.k1 = b.k1 and a.k2 > b.k2) 
-             order by isnull(k1), 1, 2, 3, 4, 5 limit 65535"""
+             order by isnull(k), 1, 2, 3, 4, 5 limit 65535"""
```
Comment (englefly) [inline regression-test/suites/nereids_p0/join/test_join.groovy:406]: why change this case? The test purpose of using k or k1 are different.

**Review by morrySnow — DISMISSED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**